store/tikv/gcworker/gc_worker.go

```diff
@@ -511,7 +526,96 @@ func resolveLocks(ctx goctx.Context, store tikv.Storage, safePoint uint64, ident
 	return nil
 }
 
-func doGC(ctx goctx.Context, store tikv.Storage, safePoint uint64, identifier string) error {
+type gcTask struct {
+	startKey  []byte
+	endKey    []byte
+	safePoint uint64
+}
+
+type gcResult struct {
+	successRegions int
+	failedRegions  int
+	lastErr        error
+}
+
+type gcTaskWorker struct {
+	store    tikv.Storage
+	taskCh   chan *gcTask
+	resultCh chan *gcResult
+	wg       *sync.WaitGroup
+}
+
+func newGCTaskWorker(store tikv.Storage, taskCh chan *gcTask, resultCh chan *gcResult, wg *sync.WaitGroup) *gcTaskWorker {
+	return &gcTaskWorker{store, taskCh, resultCh, wg}
+}
+
+func (w *gcTaskWorker) run() {
+	defer w.wg.Done()
+	for {
+		task := <-w.taskCh
+		if task == nil {
+			w.resultCh <- nil
+			return
+		}
+
+		res, err := doGCForRange(w.store, task.startKey, task.endKey, task.safePoint)
```

Comment:
It would be cleaner to define the `doGCForRange` as a method of `gcTaskWorker`

Reply:
ok